rm-datasource/src/main/java/org/apache/seata/rm/datasource/xa/ConnectionProxyXA.java

```diff
@@ -140,8 +140,12 @@ public synchronized void xaCommit(String xid, long branchId, String applicationD
      * @param applicationData application data
      */
     public synchronized void xaRollback(String xid, long branchId, String applicationData) throws XAException {
-        XAXid xaXid = XAXidBuilder.build(xid, branchId);
-        xaRollback(xaXid);
+        if (this.xaBranchXid != null) {
```

Comment:
If the XA transaction is rollback due to timeout, the xaBranchXid here is not null. Should we verify that xid and branchId in the parameter are the same xaBranchXid?

Reply:
The ConnectionProxyXA object for rollback is obtained using the xid and branchId from the parameters. Therefore, when xaBranchXid != null, the xid and branchId of xaBranchXid are the same as those in the parameters. Additional validation is only necessary if the ConnectionProxyXA is not obtained using xid and branchId.

Reply:
> If the XA transaction is rollback due to timeout, the xaBranchXid here is not null. Should we verify that xid and branchId in the parameter are the same xaBranchXid?

当前的connectionProxyXA是通过xid和branch生成的XAXid去map中进行获取，所以可以保证它们的xid和branch一定是相同的。
The current connectionProxyXA is obtained from the XAXid generated by xid and branch in the map, so it can be guaranteed that their xid and branch must be the same.